src/main/java/de/hpi/bpt/argos/storage/DatabaseAccessImpl.java

```diff
@@ -144,9 +146,51 @@ public boolean deleteArtifacts(PersistenceArtifact... artifacts) {
 	 * {@inheritDoc}
 	 */
 	@Override
-	public <R, Q> R getArtifacts(Session session, Query<Q> query, Transaction transaction, Callable<R> getValue, R defaultValue) {
+	public <ResultType, QueryType> ResultType getArtifacts(Session session,
+														   Query<QueryType> query,
+														   Transaction transaction,
+														   Callable<ResultType> getValue,
+														   ResultType defaultValue) {
+
+		ObjectWrapper<ResultType> result = new ObjectWrapper<>();
+		WatchImpl.measure("get artifacts",
+				() -> result.set(executeGetArtifacts(session, query, transaction, getValue, defaultValue)));
+
+		return result.get();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public <ResultType, ResultImplType extends ResultType> List<ResultType> getArtifactsById(Session session,
+																							 Class<ResultImplType> resultTypeClass,
+																							 Long... ids) {
+
+		ObjectWrapper<List<ResultType>> result = new ObjectWrapper<>();
+		WatchImpl.measure("get artifacts by id",
+				() -> result.set(executeGetArtifactsById(session, resultTypeClass, ids)));
+		return result.get();
+	}
+
+	/**
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AJavadoc_Style_BP2016)